### PR TITLE
Add the suite starter to an ETOS cluster

### PR DESCRIPTION
**Pull request by t-persson**
### Applicable Issues

fixes: #288 

### Description of the Change

This will enable us to run both versions of ETOS using the same ETOS cluster.

### Alternate Designs

None really.

### Possible Drawbacks

Lots of code that needs to be removed later, but the ability to use the same cluster to run both versions of ETOS is a significant improvement.

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by: Tobias Persson <tobias.persson@axis.com>

### CODE REVIEW

File: api/v1alpha1/common_types.go [hunk 1]
```diff
@@ -127,7 +127,7 @@ type RabbitMQ struct {
 	SSL string `json:"ssl"`
 	// +kubebuilder:default=/
 	// +optional
-	Vhost       string `json:"vhost"`
-	QueueName   string `json:"queueName,omitempty"`
-	QueueParams string `json:"queueParams,omitempty"`
+	Vhost string `json:"vhost"`
+	// QueueName   string `json:"queueName,omitempty"`
+	// QueueParams string `json:"queueParams,omitempty"`
```
Comment (t-persson) [inline api/v1alpha1/common_types.go:132]: Remove

**Review by andmat900 — APPROVED** [review]

**Comment by t-persson** [comment]
The routing key tag change is problematic when running in the controller environment. Must fix that before merge

**Comment by andmat900** [comment]
Regarding this:
> This will enable us to run both versions of ETOS using the same ETOS cluster.

Please describe in a README (or similar) how this separation is designed to work. Not very clear just by reading the code. It would help understand potential problems, some of which we already seen (e. g. routing tagging).

**Comment by t-persson** [comment]
> Regarding this:
> 
> > This will enable us to run both versions of ETOS using the same ETOS cluster.
> 
> Please describe in a README (or similar) how this separation is designed to work. Not very clear just by reading the code. It would help understand potential problems, some of which we already seen (e. g. routing tagging).

There is no real separation in this code, we just make sure to deploy the suite starter so that the ETOS API can send a TERCC that's being picked up and not ignored.

File: api/v1alpha1/cluster_types.go [hunk 2]
```diff
@@ -88,31 +97,96 @@ type Database struct {
 	Etcd Etcd `json:"etcd"`
 }
 
+// ETOSAPI describes the deployment of the ETOS API.
 type ETOSAPI struct {
 	Image `json:",inline"`
+	// The provider secrets are necessary in order to run ETOS the old way and not using the controller.
```
Comment (fredjn) [inline api/v1alpha1/cluster_types.go:103]: "the old way" is not a very clear way to describe the situation IMO.

```suggestion
	// The provider secrets are necessary in order to deploy and run ETOS without using kubernetes controller(s).
```

File: api/v1alpha1/cluster_types.go [hunk 3]
```diff
@@ -88,31 +97,96 @@ type Database struct {
...
+	// They can be removed from here when the suite starter is no longer in use.
```
[the remark this answers is not in the document]
Comment (t-persson) [reply api/v1alpha1/cluster_types.go:104]: I tend to do it the other way around, where I write the issue after the PR is merged and link into the code from that issue.

**Review by fredjn — APPROVED** [review]

**Pull request merged**